### ISSUE TITLE
Crash on text input

### ISSUE THREAD

**Issue by ericweiner** [EtalumaSupport/LumaViewPro#199]
(Code downloaded on 2/22/23)

Steps to reproduce:

- Click on XY on left
- Enter text in the X location (i.e. "35")
- Click on the Y location.
- Crash.

(presumably related to recent change on text inputs. issue #195 

-E

TrinamicBoard.target_pos(X) succeeded
TrinamicBoard.target_pos(Y) succeeded
ShaderViewer.on_touch_down()
Stage.on_touch_down()
Stage.on_touch_down()
XYStageControl.set_xposition()
[INFO ] [Base ] Leaving application in progress...
Traceback (most recent call last):
File "C:\Users\Eric Weiner\Downloads\LumaViewPro-master-2023-02-22\LumaViewPro-master\lumaviewpro.py", line 2877, in
LumaViewProApp().run()
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\app.py", line 955, in run
runTouchApp()
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\base.py", line 574, in runTouchApp
EventLoop.mainloop()
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\base.py", line 339, in mainloop
self.idle()
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\base.py", line 383, in idle
self.dispatch_input()
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\base.py", line 334, in dispatch_input
post_dispatch_input(*pop(0))
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\base.py", line 263, in post_dispatch_input
listener.dispatch('on_motion', etype, me)
File "kivy_event.pyx", line 731, in kivy.event.EventDispatcher.dispatch
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\core\window_init.py", line 1660, in on_motion
self.dispatch('on_touch_down', me)
File "kivy_event.pyx", line 731, in kivy.event.EventDispatcher.dispatch
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\core\window_init.py", line 1677, in on_touch_down
if w.dispatch('on_touch_down', touch):
File "kivy_event.pyx", line 731, in kivy._event.EventDispatcher.dispatch
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\uix\widget.py", line 589, in on_touch_down
if child.dispatch('on_touch_down', touch):
File "kivy_event.pyx", line 731, in kivy._event.EventDispatcher.dispatch
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\uix\widget.py", line 589, in on_touch_down
if child.dispatch('on_touch_down', touch):
File "kivy_event.pyx", line 731, in kivy._event.EventDispatcher.dispatch
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\uix\widget.py", line 589, in on_touch_down
if child.dispatch('on_touch_down', touch):
File "kivy_event.pyx", line 731, in kivy._event.EventDispatcher.dispatch
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\uix\accordion.py", line 308, in on_touch_down
return super(AccordionItem, self).on_touch_down(touch)
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\uix\widget.py", line 589, in on_touch_down
if child.dispatch('on_touch_down', touch):
File "kivy_event.pyx", line 731, in kivy._event.EventDispatcher.dispatch
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\uix\widget.py", line 589, in on_touch_down
if child.dispatch('on_touch_down', touch):
File "kivy_event.pyx", line 731, in kivy._event.EventDispatcher.dispatch
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\uix\widget.py", line 589, in on_touch_down
if child.dispatch('on_touch_down', touch):
File "kivy_event.pyx", line 731, in kivy._event.EventDispatcher.dispatch
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\uix\widget.py", line 589, in on_touch_down
if child.dispatch('on_touch_down', touch):
File "kivy_event.pyx", line 731, in kivy._event.EventDispatcher.dispatch
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\uix\widget.py", line 589, in on_touch_down
if child.dispatch('on_touch_down', touch):
File "kivy_event.pyx", line 731, in kivy._event.EventDispatcher.dispatch
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\uix\widget.py", line 589, in on_touch_down
if child.dispatch('on_touch_down', touch):
File "kivy_event.pyx", line 731, in kivy._event.EventDispatcher.dispatch
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\uix\widget.py", line 589, in on_touch_down
if child.dispatch('on_touch_down', touch):
File "kivy_event.pyx", line 731, in kivy._event.EventDispatcher.dispatch
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\uix\textinput.py", line 1541, in on_touch_down
if super().on_touch_down(touch):
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\uix\behaviors\focus.py", line 462, in on_touch_down
self.focus = True
File "kivy\properties.pyx", line 520, in kivy.properties.Property.set
File "kivy\properties.pyx", line 567, in kivy.properties.Property.set
File "kivy\properties.pyx", line 606, in kivy.properties.Property._dispatch
File "kivy_event.pyx", line 1307, in kivy._event.EventObservers.dispatch
File "kivy_event.pyx", line 1213, in kivy._event.EventObservers._dispatch
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\uix\behaviors\focus.py", line 402, in _on_focus
self._bind_keyboard()
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\uix\textinput.py", line 3010, in _bind_keyboard
super()._bind_keyboard()
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\uix\behaviors\focus.py", line 420, in _bind_keyboard
self._ensure_keyboard()
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\uix\behaviors\focus.py", line 409, in ensure_keyboard
keyboard = self.keyboard = EventLoop.window.request_keyboard(
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\core\window\window_sdl2.py", line 878, in request_keyboard
request_keyboard(
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\core\window_init.py", line 2341, in request_keyboard
self.release_keyboard(target)
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\core\window\window_sdl2.py", line 891, in release_keyboard
super(WindowSDL, self).release_keyboard(*largs)
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\core\window_init.py", line 2425, in release_keyboard
callback()
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\uix\behaviors\focus.py", line 454, in _keyboard_released
self.focus = False
File "kivy\properties.pyx", line 520, in kivy.properties.Property.set
File "kivy\properties.pyx", line 567, in kivy.properties.Property.set
File "kivy\properties.pyx", line 606, in kivy.properties.Property._dispatch
File "kivy_event.pyx", line 1307, in kivy._event.EventObservers.dispatch
File "kivy_event.pyx", line 1189, in kivy._event.EventObservers._dispatch
File "C:\Users\Eric Weiner\AppData\Roaming\Python\Python310\site-packages\kivy\lang\builder.py", line 55, in custom_callback
exec(kvlang.co_value, idmap)
File "C:\Users\Eric Weiner\Downloads\LumaViewPro-master-2023-02-22\LumaViewPro-master\lumaviewpro.kv", line 536, in
on_focus: if not self.focus: root.set_xposition(self.text)
File "C:\Users\Eric Weiner\Downloads\LumaViewPro-master-2023-02-22\LumaViewPro-master\lumaviewpro.py", line 1233, in set_xposition
stage_x, stage_y = protocol_settings.plate_to_stage(x_pos, 0)
File "C:\Users\Eric Weiner\Downloads\LumaViewPro-master-2023-02-22\LumaViewPro-master\lumaviewpro.py", line 1384, in plate_to_stage
sx = x_max - settings['stage_offset']['x']/1000 - px
TypeError: unsupported operand type(s) for -: 'float' and 'str'

**Comment by ericweiner** [comment]
Actually it crashes on any text input on the X or Y. It is also coming up with incorrect numbers here on the display.

**Comment by kmcleaster** [comment]
> Reference in new
While testing with Chris this morning, we began experiencing the exact same behavior. We were trying to get an image for testing using a 60X objective. Chris had understood that when using the Stage Controls "Set X" and "Set Y" that these values would be remembered when opening and closing LVPro. Shortly before the problem occurred, we noticed that the X coordinate   was persistent, but the Y value was not. Trying to change the Y value in the textbox cause a crash. Relaunching LVPro, the crashing behavior, identical to Eric's observations prevailed until I unplugged and re-plugged the USB connection at the PC.
